src/managers/BaseManager.js

```diff
@@ -42,12 +42,12 @@ class BaseManager {
   }
 
   add(data, cache = true, { id, extras = [] } = {}) {
-    const existing = this.cache.get(id || data.id);
-    if (existing && existing._patch && cache) existing._patch(data);
+    const existing = this.cache.get(id ?? data.id);
+    if (cache) existing?._patch(data);
```

Comment:
Not sure if there are structures with a manager that lack of a _patch method, but...
```suggestion
    if (cache) existing?._patch?.(data);
```

Reply:
all structures extend base which has an _patch method on it so ?